test/observer/permissions.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:collection';
+
+import 'package:flutter/services.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+
+import '../test.dart';
+
+/// An observer for platform permission requests.
+class PermissionsChannelObserver {
+  /// The method channel used by the flutter permissions package
+  static const MethodChannel _channel = MethodChannel('flutter.baseflow.com/permissions/methods');
+  final List<Permission> _requestedPermissions = [];  // The permissions which were requested.
+  List<Permission> get requestedPermissions  => UnmodifiableListView(_requestedPermissions);
```

Comment:
here and for other getters - move private variable under the getter and make the comment to be a doc comment of the getter

also add a newline between different getters

example:

```dart
  /// The permissions which were requested.
  List<Permission> get requestedPermissions  => UnmodifiableListView(_requestedPermissions);
  final List<Permission> _requestedPermissions = [];

  /// ....
```

Reply:
Actually, the comment is already for the getter, at least my IDE displays for the getter and not for the variable. But I agree that your suggestion is cleaner.